src/ProjectManager.cpp

```diff
@@ -92,9 +93,11 @@ const ProjectManager &ProjectManager::Get( const AudacityProject &project )
    return Get( const_cast< AudacityProject & >( project ) );
 }
 
-ProjectManager::ProjectManager( AudacityProject &project )
-   : mProject{ project }
-   , mTimer{ std::make_unique<wxTimer>(this, AudacityProjectTimerID) }
+ProjectManager::ProjectManager(AudacityProject& project)
+    : mProject { project }
+    , mTempoListener { project, TrackList::Get(project) }
+    , mTempoPublisher { project, mTempoListener }
```

Comment:
Can the listener and publisher be new attached objects on the project, so that no intrusion in ProjectManager is necessary?

Reply:
Sounds like a use case for an attachment, indeed : these blokes are just here to live, no one manipulates them ...
The attachment should still be defined in `src`, though, the only possible get-together place for `ProjectTimeSignature` and `TrackList` without modifying the dependency graph.